sql/src/main/java/io/crate/planner/Planner.java

```diff
@@ -371,6 +371,10 @@ private UpsertById processInsertStatement(InsertFromValuesAnalyzedStatement anal
             List<String> partitions = analysis.generatePartitions();
             String[] indices = partitions.toArray(new String[partitions.size()]);
             for (int i = 0; i < indices.length; i++) {
+                if (analysis.tableInfo().isAClosedIndex(indices[i])) {
+                    LOGGER.warn(String.format(Locale.ENGLISH, "Trying to insert value into closed partition: %s", indices[i]));
```

Comment:
Would it be possible to handle it wherever the insert/close/ignore case is handled? I think it's not strictly necessary to prevent this race condition but if it's easy to do, we should